src/redirects.js

```diff
@@ -22,32 +21,10 @@ export const setup = (app) => {
     res.redirect(`/groups${req.originalUrl}`);
   });
 
-  /** prior to June 2021 our core nCoV builds were available at
-   * /ncov/global, ncov/asia etc. These used GISAID data exclusively.
-   * We now have GISAID builds and GenBank builds, and so the URLs
-   * (i.e. names on s3://nextstrain-data) have changed. We add redirects
-   * for the old URLs to point to the new GISAID URLs.
-   * The timestamped URLs (e.g. /ncov/:region/YYYY-MM-DD) which currently exist
-   * will not be redirected, but new URLs will be of the format
-   * /ncov/gisaid/:region/YYYY-MM-DD.
-   */
-  app.route('/ncov/:region((global|asia|oceania|north-america|south-america|europe|africa))')
-    .get((req, res) => res.redirect(
-      url.format({
-        pathname: `/ncov/gisaid/${req.params.region}`,
-        query: req.query
-      })
-    ));
-
-  /**
-   * We shifted from using 'monkeypox' to 'mpox', as per WHO naming
-   * recommendations. Note that monkeypox YYYY-MM-DD URLs remain,
-   * e.g. /monkeypox/hmpxv1/2022-09-04
-   */
-  app.route('/monkeypox').get((req, res) => res.redirect('/mpox'));
-  app.route('/monkeypox/mpxv').get((req, res) => res.redirect('/mpox/all-clades'));
-  app.route('/monkeypox/hmpxv1').get((req, res) => res.redirect('/mpox/clade-IIb'));
-  app.route('/monkeypox/hmpxv1/big').get((req, res) => res.redirect('/mpox/lineage-B.1'));
+  /* handle redirects for dataset paths which have changed name */
+  for (const [requestPath, redirectPath] of datasetRedirects()) {
+    app.route(requestPath).get((req, res) => res.redirect(redirectPath));
```

Comment:
_non-blocking_

This doesn't preserve the query of `requestPath`, which is a behaviour change for the `/ncov/…` redirects. Is that intentional? The query was preserved for a reason (see b0d7398ab4aa717c51b4dcca44e5d51a7e688f48 and https://github.com/nextstrain/nextstrain.org/pull/377), though it's not clear if the reason still applies/matters given the passage of time.

Reply:
Huh. That's no good. I will amend the commit so that queries are preserved across the redirects and also add this to the monkeypox redirects (e.g. `nextstrain.org/monkeypox?c=region` should preserve the query).

Reply:
I've fixed this up, but there's a huge caveat -- here's the salient part of the commit message:

URL queries are now preserved for these redirects for all datasets. This
is a change in behaviour for the monkeypox URLs - as an example
"nextstrain.org/monkeypox/mpxv?c=region" will now redirect to
"nextstrain.org/mpox/all-clades?c=region". There is a very sizeable
caveat to this however, because if the redirect path described here is
subsequently redirected by our `canonicalizeDataset` middleware the
query will be discarded at that point. In practice, all of the ncov
paths and 1/4 of the mpox paths described here are subsequently
redirected and so the URL query is not preserved.
<https://github.com/nextstrain/nextstrain.org/issues/792> tracks this.